cmd/zc_traverser_blob.go

```diff
@@ -162,18 +164,33 @@ func (t *blobTraverser) Traverse(preprocessor objectMorpher, processor objectPro
 	blobProperties, isBlob, isDirStub, propErr := t.getPropertiesIfSingleBlob()
 
 	if stgErr, ok := propErr.(azblob.StorageError); ok {
+
 		// Don't error out unless it's a CPK error just yet
 		// If it's a CPK error, we know it's a single blob and that we can't get the properties on it anyway.
 		if stgErr.ServiceCode() == common.CPK_ERROR_SERVICE_CODE {
-			return errors.New("this blob uses customer provided encryption keys (CPK). At the moment, AzCopy does not support CPK-encrypted blobs. " +
-				"If you wish to make use of this blob, we recommend using one of the Azure Storage SDKs")
+			if t.getProperties {
+				return errors.New("this blob uses customer provided encryption keys (CPK). At the moment, AzCopy does not support CPK-encrypted blobs. " +
+					"If you wish to make use of this blob, we recommend using one of the Azure Storage SDKs")
+			} else {
+				blobProperties = &azblob.BlobGetPropertiesResponse{}
+				isBlob = true
+				isDirStub = true
+				propErr = nil
+			}
 		}
 
 		if resp := stgErr.Response(); resp == nil {
 			return fmt.Errorf("cannot list files due to reason %s", stgErr)
 		} else {
 			if resp.StatusCode == 403 { // Some nature of auth error-- Whatever the user is pointing at, they don't have access to, regardless of whether it's a file or a dir stub.
-				return fmt.Errorf("cannot list files due to reason %s", stgErr)
+				if t.getProperties {
```

Comment:
Here error is not related to CPK and might be some other auth error, do we need to return dummy properties here?